lib/modules/manager/nuget/util.ts

```diff
@@ -210,7 +211,10 @@ function sortPatterns(
     return -1;
   }
 
-  return a[0].localeCompare(b[0]) * -1;
+  const aTrim = a[0].replace(/\*$/, '');
+  const bTrim = b[0].replace(/\*$/, '');
```

Comment:
regex is too expensive, use slice


```suggestion
  const aTrim = a[0].slice(0, -1);
  const bTrim = b[0].slice(0, -1);
```